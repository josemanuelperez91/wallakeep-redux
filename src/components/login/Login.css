.Login {
  padding: 10px;
  max-width: 480px;
  width: 100%;
}
.Login form {
}
