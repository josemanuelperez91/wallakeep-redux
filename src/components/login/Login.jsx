import React from 'react';
import { withRouter, Link } from 'react-router-dom';
import { signIn } from '../../js/apiCalls';
class Login extends React.Component {
  constructor() {
    super();
    this.state = {
      username: '',
      password: ''
    };
  }

  handleInput = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  handleSubmit = event => {
    event.preventDefault();

    signIn({ ...this.state }).then(response => {
      if (response.ok) {
        this.props.history.push('/home');
      }
    });
  };
  render() {
    return (
      <div className="Login">
        <h1>Login</h1>
        <form onSubmit={this.handleSubmit}>
          <input
            name="username"
            onChange={this.handleInput}
            placeholder="username"
            type="text"
          />
          <input
            autoComplete="password"
            name="password"
            onChange={this.handleInput}
            placeholder="password"
            type="password"
          />
          <button className="greenButton" type="submit">
            Sign In
          </button>
          <button type="button">
            <Link to="/register">I dont have an accout</Link>
          </button>
        </form>
      </div>
    );
  }
}
export default withRouter(Login);
