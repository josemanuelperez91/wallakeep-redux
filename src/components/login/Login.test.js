import React from 'react';
import { mount, shallow } from 'enzyme';
import Login from './Login';
import { MemoryRouter } from 'react-router-dom';

describe('Snapshot component', () => {
  describe('Login', () => {
    const props = {
      signIn: jest.fn(),
    };

    it('should test snapshot', () => {
      const wrapper = shallow(<Login {...props} />);
      expect(wrapper).toMatchSnapshot();
    });

    const username = 'testuser';
    const password = 'testpass';

    it('Should perform a signIn action', () => {
      const wrapper = mount(
        <MemoryRouter>
          <Login {...props} />
        </MemoryRouter>
      );

      const form = wrapper.find('form');
      const usernameInput = wrapper.find('input').at(0);
      const passwordInput = wrapper.find('input').at(1);

      usernameInput.instance().value = username;
      passwordInput.instance().value = password;

      usernameInput.simulate('change');
      passwordInput.simulate('change');

      form.simulate('submit');
      expect(props.signIn).toHaveBeenCalledWith({
        username,
        password,
      });
    });
  });
});
