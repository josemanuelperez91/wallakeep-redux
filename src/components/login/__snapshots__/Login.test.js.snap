// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot component Login should test snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
